aioesphomeapi/connection.py

```diff
@@ -131,6 +132,46 @@ class ConnectionState(enum.Enum):
 CONNECTION_STATE_CLOSED = ConnectionState.CLOSED
 
 
+def _make_hello_request(client_info: str) -> HelloRequest:
+    """Make a HelloRequest."""
+    return HelloRequest(
+        client_info=client_info,
+        api_version_major=1,
+        api_version_minor=9,
+    )
+
+
+_cached_make_hello_request = lru_cache(maxsize=16)(_make_hello_request)
+make_hello_request = _cached_make_hello_request
+
+
+def _handle_timeout(fut: asyncio.Future[None]) -> None:
+    """Handle a timeout."""
+    if not fut.done():
+        fut.set_exception(asyncio_TimeoutError)
+
+
+handle_timeout = _handle_timeout
+
+
+def _handle_complex_message(
+    fut: asyncio.Future[None],
+    responses: list[message.Message],
+    do_append: Callable[[message.Message], bool] | None,
+    do_stop: Callable[[message.Message], bool] | None,
```

Comment:
It was a bit nicer when they were bound to the object, but there is no self use so they technically should be separate methods.